src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,107 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int INITIAL_SIZE = 10;
+    private Object[] array;
+    private int size;
+
+    public ArrayList() {
+        this.array = new Object[INITIAL_SIZE];
+        this.size = 0;
+    }
+
+    public ArrayList(int size) {
+        if (size < 0) {
+            throw new RuntimeException("Provide size less than 0");
+        } else if (size == 0) {
+            this.array = new Object[INITIAL_SIZE];
+        } else {
+            this.array = new Object[size];
+        }
+        this.size = 0;
+    }
+
+    private void grow() {
+        Object[] resizedArray = new Object[(array.length * 3) / 2];
+        System.arraycopy(array, 0, resizedArray, 0, array.length);
+        this.array = resizedArray;
+    }
+
+    private boolean isIndexInBound(int index) {
+        return index < size && index >= 0;
+    }
 
     @Override
     public void add(T value) {
-
+        add(value, size);
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (size == array.length - 1) {
```

Comment:
```suggestion
        if (size == array.length) {
```
there still will be one empty slot if size == array.length - 1